packages/react-dom/src/server/ReactPartialRendererHooks.js

```diff
@@ -473,6 +474,12 @@ function useTransition(
   return [startTransition, false];
 }
 
+let serverId: number = 0;
+
+function useOpaqueReference(): string | IdObject {
+  return 's_' + (serverId++).toString();
```

Comment:
Let's use `toString(36)` here. It'll save us a couple of bytes when we have many IDs.